File: src/main/java/infinitespire/relics/crystals/EmpoweringShard.java
package infinitespire.relics.crystals;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.powers.StrengthPower;
import infinitespire.InfiniteSpire;
import infinitespire.abstracts.InfiniteRelic;

public class EmpoweringShard extends InfiniteRelic{

    public static final String ID = InfiniteSpire.createID("EmpoweringShard");

    public EmpoweringShard(){
        super(ID, "empoweringshard", RelicTier.SPECIAL, LandingSound.CLINK);
    }

    @Override
    public void atBattleStart() {
        AbstractDungeon.actionManager.addToBottom(
            new ApplyPowerAction(
                AbstractDungeon.player,
                AbstractDungeon.player,
                new StrengthPower(AbstractDungeon.player, this.counter), this.counter));
    }
}

File: src/main/kotlin/com/blanktheevil/infinitespire/relics/EmpoweringShard.kt
package com.blanktheevil.infinitespire.relics

import com.blanktheevil.infinitespire.extensions.applyPower
import com.blanktheevil.infinitespire.extensions.makeID
import com.blanktheevil.infinitespire.extensions.player
import com.blanktheevil.infinitespire.relics.abstracts.CrystalRelic
import com.megacrit.cardcrawl.powers.StrengthPower

class EmpoweringShard : CrystalRelic(ID, IMG) {
  companion object {
    val ID = "EmpoweringShard".makeID()
    private const val IMG = "empoweringshard"
  }

  override fun atBattleStart() {
    player.applyPower(
      StrengthPower(player, counter),
      amount = counter
    )
  }
}